fix github pages deploy guide (#1388)
withastro/docs: src/pages/en/guides/deploy/github.md

Astro maintains the official `withastro/action` to deploy your project with very little configuration. Follow the instructions below to deploy your Astro site to GitHub pages, and see [the package README](https://github.com/withastro/action) if you need more information.

1. Set the [`site`](/en/reference/configuration-reference/#site) and, if needed, [`base`](/en/reference/configuration-reference/#base) options in `astro.config.mjs`.

>     - `site` should be `https://<YOUR_USERNAME>.github.io` or `https://my-custom-domain.com`

Reviewer: ``````suggestion
    
```astro title="astro.config.mjs"
import { defineConfig } from 'astro/config'

export default defineConfig({
  site: 'https://astronaut.github.io'
  base: '/my-repo'
})
```
    - `site` should be `https://<YOUR_USERNAME>.github.io` or `https://my-custom-domain.com`
``````

Trying this here to see how it looks *before* the explanation.

Astro maintains the official `withastro/action` to deploy your project with very little configuration. Follow the instructions below to deploy your Astro site to GitHub pages, and see [the package README](https://github.com/withastro/action) if you need more information.

1. Set the [`site`](/en/reference/configuration-reference/#site) and, if needed, [`base`](/en/reference/configuration-reference/#base) options in `astro.config.mjs`.
    

>     ```js title="astro.config.mjs"

Author: highlight base and site options
```suggestion
    ```js title="astro.config.mjs" ins={4-5}
```

Reviewer: I debated this myself, but sure, let's do it!

    import { defineConfig } from 'astro/config'

    export default defineConfig({

>       site: 'https://astronaut.github.io'
>       base: '/my-repo'

Reviewer: ```suggestion
      site: 'https://astronaut.github.io',
      base: '/my-repo',
```

    })
    ```
    - `site` should be `https://<YOUR_USERNAME>.github.io` or `https://my-custom-domain.com`
    - `base` should be your repository’s name starting with a forward slash, for example `/my-repo`. This is so that Astro understands your website's root is `/my-repo`, rather than the default `/`.
    
    :::note

>       Don't set a `base` parameter if:

Author: I am trying to sound a bit more friendly here 😇
```suggestion
      You don't need to set a `base` if:
```

Reviewer: Haha, sometimes you don't need to be friendly -- you need to make sure they DON'T DO something! I would honestly keep as is!

Astro maintains the official `withastro/action` to deploy your project with very little configuration. Follow the instructions below to deploy your Astro site to GitHub pages, and see [the package README](https://github.com/withastro/action) if you need more information.

1. Set the [`site`](/en/reference/configuration-reference/#site) and, if needed, [`base`](/en/reference/configuration-reference/#base) options in `astro.config.mjs`.
    

> ```astro title="astro.config.mjs"

Reviewer: ```suggestion
```js title="astro.config.mjs"
```

1. Set the [`site`](/en/reference/configuration-reference/#site) and, if needed, [`base`](/en/reference/configuration-reference/#base) options in `astro.config.mjs`.
    - `site` should be something like `https://<YOUR_USERNAME>.github.io`
    - `base` should be your repository’s name starting with a forward slash, for example `/my-repo`. This is so that Astro understands your website's root is "/my-repo", rather than default "/".
    
    :::note
    Don't set a `base` parameter if:

>     - your repository itself is called <YOUR_USERNAME>.github.io

Author: ```suggestion
    - your repository itself is named <YOUR_USERNAME>.github.io
```

1. Set the [`site`](/en/reference/configuration-reference/#site) and, if needed, [`base`](/en/reference/configuration-reference/#base) options in `astro.config.mjs`.

>     - `site` should be something like `https://<YOUR_USERNAME>.github.io`

Reviewer: I know this was existing text when you got here, but "should be something like" without more explicit instruction makes me nervous. 😅 

I think I recall from earlier versions of this page that you SHOULD be using `https://<YOUR_USERNAME>.github.io` -- but, is it possible to deploy TWO DIFFERENT sites on GitHub pages? What would you do then, since you only have one username.

I think it might also be helpful to be more DIRECT about saying to use this for `site` (like, just use this!) or to add what another "something like" could be, for example in the case of having two sites deployed. (Is that even possible?)

Reviewer 2: Multiple sites are possible, and as far as I can tell, the `site` for them is the same. `sarah11918.github.io/blog` and `sarah11918.github.io/portfolio` both share a site of `sarah11918.github.io`.

The other "something like" is when you use a custom domain. Then it's the custom domain you fill in as `site`.

Reviewer: Ah, good call, that's where base comes in !

I thought about also suggesting that we show a custom domain as a possiblility there, but then our tip for "Want to use a custom domain?" is alllll the way down at the bottom. And, THAT doesn't say to update SITE, so I didn't know whether that cname file would actually just "overwrite" a github.io site, or whether you were supposed to update SITE, too.

So then, what about something like:

```
- `site` should be `https://<YOUR_USERNAME>.github.io` or `https://my-custom-domain.com`
```

Author: You are right, we forgot to include an update site in that little tip 😅. my best guess is that this file sets an alias for `username.github.io` but the build is using site, whoops 😇

Author: ```suggestion
    - `site` be should be `https://<YOUR_USERNAME>.github.io` or `https://my-custom-domain.com`
```

    - `base` should be your repository’s name starting with a forward slash, for example `/my-repo`. This is so that Astro understands your website's root is `/my-repo`, rather than the default `/`.

>     

Reviewer: Is it too much to show an `astro.config.mjs` example here with those fields filled out?

Reviewer 2: @odwrotnie was asking for an example in #1376. They managed to [misplace the parameters](https://github.com/instap-app/website-definition/blob/4fd94706505c385128284a3f0b27eb85ff2d85ca/astro.config.mjs#L9-L10) outside of the defineConfig object, because they didn't find this clear enough.

    :::note
    Don't set a `base` parameter if:

>     - your repository is named `<YOUR_USERNAME>.github.io`
>     - you're using a custom domain

Reviewer 3: We tend to go with the AP Stylebook recommendation for bullet points, I would change this for consistency:
```suggestion
    - Your repository is named `<YOUR_USERNAME>.github.io`.
    - You're using a custom domain.
```

    :::

    :::caution

>     If you are setting up `base` only to deploy to GitHub do not forget to update your routes to include your `base`.

Reviewer: ```suggestion
        If you did not previously have a value for `base` set, and are only configuring this value so that you can deploy to GitHub, you must update your internal page links to now include your `base`.
```
What do you (all!) think about something more "positively" phrased (you must do this), rather than "Don't forget  to!" which may or may not have been clear to the reader before?

>     If your repository is named `<YOUR_USERNAME>.github.io`, you don’t need to include `base`.

Author: ```suggestion
    If your repository has the same name as your username it will be deployed in `<YOUR_USERNAME>.github.io`, hence, you don’t need to include a `base`.
```

Author: @swift502 what do you think about this reword?

Reviewer 4: Shouldn't the same apply for custom domains?

Reviewer 2: @kevinzunigacuellar Ohhhkay, I think I get it. Thanks for being patient with me.

@fredoist is right. You must _not_ include the base if you're using a custom domain, otherwise assets fail to load. I just tested that on my website.

I'd be much more explicit, because it's so convoluted:

> If you're using GitHub's default site url, e.g. `<YOUR_USERNAME>.github.io/my-repo`, set the `base` parameter to your repository’s name starting with a forward slash, for example `/my-repo`. This is so that Astro understands your website's root is "/my-repo", rather than default "/".
> 
> __Don't__ set the `base` parameter if:
> 
> - your repository itself is called `<YOUR_USERNAME>.github.io`
> - you're using a custom domain

---

Combined with the "update your routes" caution box you added, it feels appropriately comprehensive.
Maybe it's a lot of text though. I'll leave it up to you.

Btw:
> If your repository has the same name as your username

I'm not sure that's a good wording because it implies swift502/swift502, which would be the special user repository, not swift502.github.io right?

Author: Yes, my mistake here 😅. I decided to keep the part of the original message and add a little more explanation for why and when you need to set a base.

Reviewer 4: LGTM 👍🏼

Reviewer 2: Yeah I like it.
